release_notes/v2_vs_v3.adoc

```diff
@@ -0,0 +1,107 @@
+
+From a packaging perspective, an image performs more tasks than a cartridge, and provides better encapsulation and flexibility. However, cartridges also included logic for building, deploying, and routing, which do not exist in images. In OpenShift v3, these additional needs are met by link:../architecture/core_concepts/builds_and_image_streams.html#source-build[Source-to-Image (S2I)] and link:../architecture/core_concepts/templates.html[template configuration].
+
+*Dependencies*
+
+In OpenShift v2, cartridge dependencies were defined with `*Configure-Order*` or `*Requires*` in
+a cartridge manifest. OpenShift v3 uses a declarative model where
+link:../architecture/core_concepts/pods_and_services.html#pods[pods] bring
+themselves in line with a predefined state. Explicit dependencies that are
+applied are done at runtime rather than just install time ordering.
+
+For example, you might require another service to be available before you start.
+Such a dependency check is always applicable and not just when you create the
+two components. Thus, pushing the dependency into a runtime dependency enables
+the system to stay healthy over time.
+
+*Collection*
+
+Whereas cartridges in OpenShift v2 were collocated within gears, link:../architecture/core_concepts/containers_and_images.html#docker-images[images] in OpenShift v3 are mapped 1:1 with link:../architecture/core_concepts/containers_and_images.html#containers[containers], which use link:../architecture/core_concepts/pods_and_services.html#pods[pods] as their collocation mechanism.
+
+*Source Code*
+
+In OpenShift v2, applications were required to have at least one web framework with one git repository. In OpenShift v3, you can choose which images are built from source and that source can be located outside of OpenShift itself. Because the source is disconnected from the images, the choice of image and source are distinct operations with source being optional.
+
+*Build*
+
+In OpenShift v2, builds occurred in place of your application gears. This meant downtime for non-scaled applications due to resource constraints. In v3, link:../architecture/core_concepts/builds_and_image_streams.html#builds[builds] happen in separate containers. Also, OpenShift v2 build results were rsynced between gears. In v3, build results are first committed as an immutable image and published to an internal registry. That image is then available to launch on any of the nodes in the cluster, or available to rollback to at a future date. This means that builds are no longer part of the cartridges and are provided as distinct choices.
```

Comment:
The last sentence confuses me:
- IIUC "cartridge" as a concept is no longer valid in v3, so "are no longer part" is redundant regardless of "builds" location / design / implementation.
- "distinct choices" sounds incomplete.